### ISSUE TITLE
how to chat with entire vault?

### ISSUE THREAD

**Issue by iamrobmat** [logancyang/obsidian-copilot#208]
Is there an option to chat not only with the active note but entire vaults?

**Comment by logancyang** [comment]
hi @iamrobmat! This mode is coming soon, still need to iron out some issues with the local RAG.

**Comment by logancyang** [comment]
> Hey, I'm willing to help out with that feature, just found out the copilot exists and want to use it but unfortunately, it's not very useful without it (a.k.a. not significantly better than the https://chat.openai.com/ which I'm already paying for)
> 
> Specifically for my use case I am willing to explicitly scope down the context based on tags mentioned in my documents and/or a list of folders in the tree.
> 
> I think with that it should be possible to rollout without additional weight of true RAG / embeddings and just stuff everything into the same prompt Arguably RAG might make the results significantly worse, depending on the implementation (for models that support big token windows)
> 
> If you don't have any strong opinion against it I can try to think of something and hack it around, hopefully not interrupting your progress with RAG

Hey @Noobgam these are great points. I will have new modes in the next revamp: 

- A Chat mode (currently called Conversation mode) with the ability to just stuff multiple notes into the same prompt for long context models. The result will be better than just RAG for less specific questions as you mentioned.
- A QA mode powered by local hybrid search (RAG + full-text), this is better for specific questions on a large collection of notes.

Sorry it's taking a while, I will ship the first one first and the local hybrid search will come after.

**Comment by peterlionelnewman** [comment]
If you were unaware, the recent addition of openai APIs to ollama should make integration more straightforward for all future work. I beleive that this should include support for embeddings within ollama shortly.

https://ollama.com/blog/openai-compatibility

**Comment by logancyang** [comment]
@peterlionelnewman I'm monitoring their progress. Ollama currently doesn't support BERT style embeddings which are required for RAG because they were waiting on BERT support from llama.cpp. Looks like a PR was just merged there https://github.com/ggerganov/llama.cpp/pull/5423. Hopefully it will be available in Ollama soon.

**Comment by logancyang** [comment]
v2.5.0 released
